### PR TITLE
fixed average rating precision in stars

### LINKED ISSUE #120: FRONTEND: fix average stars calculations + appearance

### PR DESCRIPTION
**Pull request by shaaronl**
image

Screen Shot 2024-05-28 at 10 20 43 PM

The avg stars should now reflect the avg rating more precisely. Currently when there are  no reviews, it will just show a row of black star frames, as seen in image above, but I can change that to something else. Any suggestions?

### CODE REVIEW

File: packages/react-frontend/src/routes/AlbumInfo.jsx [hunk 1]
```diff
@@ -126,7 +99,7 @@ export default function AlbumInfo() {
               {album.release_date.split("-")[0] + " "}
               {artist.artist_name} |{"  "}
               <span className="rating-stars">
-                {starRating}{" "}
+                <Rating name="album-rating" value={parseFloat(avgRating)} precision={0.5}/>
```
Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]: Hey looking at the materials UI documentation. The precision could be less than 0.5. Like 0.1 for example and then the stars would more accurately display the result you're looking for. Keeping it at 0.5 only makes the stars display in half amount of increments. https://mui.com/material-ui/react-rating/

**Review by tammy-si — COMMENTED** [review]
Okay but impovements could be made very easily because materials UI has a lot of options and resources available to make a better looking effect.

File: packages/react-frontend/src/routes/AlbumInfo.jsx (hunk 1 above)
Comment (saravoong) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]: It looks good, but when I hover over the average star rating, it changes to where my mouse is. Is it possible to make it read-only?

**Review by saravoong — COMMENTED** [review]
Look at my other comment, but also, when I was testing the code, the reviews itself doesn't show any half-stars, only full/empty stars, so it's not really precise, is it possible to show half-stars?

**Comment by BraedenAlonge** [comment]
Looks amazing Sharon! Can I suggest that, instead of the long spotify url, we can make the words "Spotify Link" into an actual clickable link? I think it would look really clean! 
Also, maybe instead of "Popularity" it could be something like "Views", since popularity seems a bit vague.

**Comment by BraedenAlonge** [comment]
Also, I tthink the Popularity" Header is a little bit vague. What about something like "Views"?

File: packages/react-frontend/src/routes/AlbumInfo.jsx (hunk 1 above)
Comment (shaaronl) [reply packages/react-frontend/src/routes/AlbumInfo.jsx]: ohhh good catch! thank you tammy, i'll look into changing that.

Comment (shaaronl) [reply packages/react-frontend/src/routes/AlbumInfo.jsx]: oh wow, i didn't even notice it could hover like that haha, yes i think i should be able to change it to read-only and i will look into this a bit more. thanks for catching this, sara!

Comment (shaaronl) [reply packages/react-frontend/src/routes/AlbumInfo.jsx:102]: also yes, i forgot to implement the precision in the individual reviews so ill change the individual reviews as well. I mainly focused on the one getting displayed in the title, but i agree this should be changed.

**Comment by shaaronl** [comment]
> Also, I tthink the Popularity" Header is a little bit vague. What about something like "Views"?

oh i think this feature was implemented by sara for testing purposes, so i don't think we'll be using it anymore because it's not actually an accurate representation of views, i think its just a popularity score?

**Comment by shaaronl** [comment]
> Look at my other comment, but also, when I was testing the code, the reviews itself doesn't show any half-stars, only full/empty stars, so it's not really precise, is it possible to show half-stars?

yes, i haven't implemented the precision in the individual reviews yet, so ill change the individual reviews as well. I mainly focused on the one getting displayed in the title, but i agree this should be changed.

**Comment by shaaronl** [comment]
> Looks amazing Sharon! Can I suggest that, instead of the long spotify url, we can make the words "Spotify Link" into an actual clickable link? I think it would look really clean!

I believe the link is already clickable...is it not working on ur end? and yes i can change the link to say 'Spotify Link' instead!

**Comment by BraedenAlonge** [comment]
> > Looks amazing Sharon! Can I suggest that, instead of the long spotify url, we can make the words "Spotify Link" into an actual clickable link? I think it would look really clean!
> 
> I believe the link is already clickable...is it not working on ur end? and yes i can change the link to say 'Spotify Link' instead!

Yes, the link is clickable! I just meant that the url  (ex. https://open.spotify.com/album/07w0rG5TETcyihsEIZR3qG) is a bit clunky. I think that the actual link should be the words "Spotify Link" to ake it look cleaner...just a suggestion though!

**Comment by shaaronl** [comment]
> > > Looks amazing Sharon! Can I suggest that, instead of the long spotify url, we can make the words "Spotify Link" into an actual clickable link? I think it would look really clean!
> > 
> > 
> > I believe the link is already clickable...is it not working on ur end? and yes i can change the link to say 'Spotify Link' instead!
> 
> Yes, the link is clickable! I just meant that the url (ex. https://open.spotify.com/album/07w0rG5TETcyihsEIZR3qG) is a bit clunky. I think that the actual link should be the words "Spotify Link" to ake it look cleaner...just a suggestion though!

ohhh okok, but i agree i think it looks better just as "Spotify Link"

**Review by Yakkaya — COMMENTED** [review]
It looks good to me! I think it could be beneficial to write unit tests, since there are more complicated functions for the average star function that require doing math, in order to ensure there's no weirdness with that/

**Comment by shaaronl** [comment]
> It looks good to me! I think it could be beneficial to write unit tests, since there are more complicated functions for the average star function that require doing math, in order to ensure there's no weirdness with that/

thanks! and yea i agree, unit tests are definitely needed here to ensure correct output for all cases + edge cases.

**Comment by saravoong** [comment]
Is it possible for you to change the color of the outline or the fill in of the average blank stars because it's bit hard to see them, like make it lighter. Or change it to the way the review blank stars look like.

**Comment by shaaronl** [comment]
hi! i just fixed the precision in the average stars displayed as well as the spotify link. take a look and lmk what yall think!

Screen Shot 2024-05-30 at 11 27 55 AM
Screen Shot 2024-05-30 at 11 27 13 AM

**Comment by shaaronl** [comment]
> Is it possible for you to change the color of the outline or the fill in of the average blank stars because it's bit hard to see them, like make it lighter. Or change it to the way the review blank stars look like.

yes i agree, ill look into changing that!

**Comment by saravoong** [comment]
I think it looks good to me! Ready to merge?

File: packages/react-frontend/src/routes/AlbumInfo.jsx [hunk 2]
```diff
@@ -145,11 +118,9 @@ export default function AlbumInfo() {
               reviews.map((review) => (
                 <div key={review._id} className="review">
                   <span className="rating-stars">
```
Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]: Unfortunately because I was unable to communicate well, we have a little merge conflict here. We could probably fix it by either deleting these lines here. or trying to combine it with the one I have merged into develop. Again my bad.

Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]:
```suggestion
```

File: packages/react-frontend/src/routes/AlbumInfo.jsx [hunk 3]
```diff
@@ -145,11 +118,9 @@ export default function AlbumInfo() {
...
-                    {Array.from({ length: 5 }, (_, i) =>
-                      i < review.rating ? "★" : "☆"
-                    ).join("")}
+                    <Rating value={parseFloat(review.rating)} precision={0.1} readOnly/>
```
Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]:
```suggestion
```

File: packages/react-frontend/src/routes/AlbumInfo.jsx [hunk 4]
```diff
@@ -145,11 +118,9 @@ export default function AlbumInfo() {
...
                   </span>
```
Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]:
```suggestion
```

File: packages/react-frontend/src/routes/AlbumInfo.jsx [hunk 5]
```diff
@@ -145,11 +118,9 @@ export default function AlbumInfo() {
...
-                  <p>{review.content}</p>
+                  <p className = "review">{review.content}</p>
```
Comment (tammy-si) [inline packages/react-frontend/src/routes/AlbumInfo.jsx]:
```suggestion
```

**Review by tammy-si — COMMENTED** [review]
Trying to help you fix some merge conflicts. Again if you choose to instead of just deleting some parts of your work, and instead get like a hybrid version of both our work combined that'd be okay too.
